sympy/printing/llvmjitcode.py

```diff
@@ -0,0 +1,192 @@
+from __future__ import print_function, division
+
+'''
+Use llvmlite to create executable functions from Sympy expressions
+
+This module requires llvmlite (https://github.com/numba/llvmlite).
+'''
+
+import ctypes
+
+from sympy.external import import_module
+from sympy.printing.printer import Printer
+from sympy import S
+from sympy.utilities.decorator import doctest_depends_on
+
+llvmlite = import_module('llvmlite')
+if llvmlite:
+    ll = import_module('llvmlite.ir').ir
+    llvm = import_module('llvmlite.binding').binding
+    llvm.initialize()
+    llvm.initialize_native_target()
+    llvm.initialize_native_asmprinter()
+
+
+class LLVMJitPrinter(Printer):
```

Comment:
The IR is converted to a string before assembly (the line `strmode = str(module)` in `llvm_jit_code`).  The question is how to give users access.

The `llvm_jit_code` function contains a lot of LLVM setup and management code.  My preference would be to keep this code out of the LLVMJITPrinter class (as future features are going to increase the size of this class anyway). 
Perhaps the `llvm_jit_code` should be converted into a class, with smaller methods, and would allow access to the string representation of the IR.